yamldecode cannot parse string started with "0b"

### Terraform Version
$ terraform1218 -v
Terraform v0.12.18

### Terraform Configuration Files
```hcl
locals {
  ci = yamldecode("key: 0ba")
}
output "ci" {
  value = local.ci
}
```


### Debug Output
```
$ terraform1218 apply
2020/01/02 14:42:32 [INFO] Terraform version: 0.12.18  
2020/01/02 14:42:32 [INFO] Go runtime version: go1.12.13
2020/01/02 14:42:32 [INFO] CLI args: []string{"/usr/local/bin/terraform1218", "apply"}
2020/01/02 14:42:32 [DEBUG] Attempting to open CLI config file: /home/wjiang/.terraformrc
2020/01/02 14:42:32 Loading CLI configuration from /home/wjiang/.terraformrc
2020/01/02 14:42:32 [DEBUG] checking for credentials in "/home/wjiang/.terraform.d/plugins"
2020/01/02 14:42:32 [INFO] CLI command args: []string{"apply"}
2020/01/02 14:42:32 [TRACE] Meta.Backend: no config given or present on disk, so returning nil config
2020/01/02 14:42:32 [TRACE] Meta.Backend: backend has not previously been initialized in this working directory
2020/01/02 14:42:32 [DEBUG] New state was assigned lineage "0748db86-469c-fd83-a9e6-511554c8e0fa"
2020/01/02 14:42:32 [TRACE] Meta.Backend: using default local state only (no backend configuration, and no existing initialized backend)
2020/01/02 14:42:32 [TRACE] Meta.Backend: instantiated backend of type <nil>
2020/01/02 14:42:32 [DEBUG] checking for provider in "."
2020/01/02 14:42:32 [DEBUG] checking for provider in "/usr/local/bin"
2020/01/02 14:42:32 [DEBUG] checking for provider in "/home/wjiang/.terraform.d/plugins"
2020/01/02 14:42:32 [WARN] found legacy provider "terraform-provider-libvirt"
2020/01/02 14:42:32 [DEBUG] found valid plugin: "libvirt", "0.0.0", "/home/wjiang/.terraform.d/plugins/terraform-provider-libvirt"
2020/01/02 14:42:32 [DEBUG] checking for provisioner in "."
2020/01/02 14:42:32 [DEBUG] checking for provisioner in "/usr/local/bin"
2020/01/02 14:42:32 [DEBUG] checking for provisioner in "/home/wjiang/.terraform.d/plugins"
2020/01/02 14:42:32 [INFO] Failed to read plugin lock file .terraform/plugins/linux_amd64/lock.json: open .terraform/plugins/linux_amd64/lock.json: no such file or directory
2020/01/02 14:42:32 [TRACE] Meta.Backend: backend <nil> does not support operations, so wrapping it in a local backend
2020/01/02 14:42:32 [INFO] backend/local: starting Apply operation
2020/01/02 14:42:32 [TRACE] backend/local: requesting state manager for workspace "default"
2020/01/02 14:42:32 [TRACE] backend/local: state manager for workspace "default" will:
 - read initial snapshot from terraform.tfstate
 - write new snapshots to terraform.tfstate
 - create any backup at terraform.tfstate.backup
2020/01/02 14:42:32 [TRACE] backend/local: requesting state lock for workspace "default"
2020/01/02 14:42:32 [TRACE] statemgr.Filesystem: preparing to manage state snapshots at terraform.tfstate
2020/01/02 14:42:32 [TRACE] statemgr.Filesystem: no previously-stored snapshot exists
2020/01/02 14:42:32 [TRACE] statemgr.Filesystem: locking terraform.tfstate using fcntl flock
2020/01/02 14:42:32 [TRACE] statemgr.Filesystem: writing lock metadata to .terraform.tfstate.lock.info
2020/01/02 14:42:32 [TRACE] backend/local: reading remote state for workspace "default"
2020/01/02 14:42:32 [TRACE] statemgr.Filesystem: reading latest snapshot from terraform.tfstate
2020/01/02 14:42:32 [TRACE] statemgr.Filesystem: snapshot file has nil snapshot, but that's okay
2020/01/02 14:42:32 [TRACE] statemgr.Filesystem: read nil snapshot
2020/01/02 14:42:32 [TRACE] backend/local: retrieving local state snapshot for workspace "default"
2020/01/02 14:42:32 [TRACE] backend/local: building context for current working directory
2020/01/02 14:42:32 [TRACE] terraform.NewContext: starting
2020/01/02 14:42:32 [TRACE] terraform.NewContext: resolving provider version selections
2020/01/02 14:42:32 [TRACE] terraform.NewContext: loading provider schemas
2020/01/02 14:42:32 [TRACE] terraform.NewContext: complete
2020/01/02 14:42:32 [TRACE] backend/local: finished building terraform.Context
2020/01/02 14:42:32 [TRACE] backend/local: requesting interactive input, if necessary
2020/01/02 14:42:32 [TRACE] Context.Input: Prompting for provider arguments
2020/01/02 14:42:32 [TRACE] backend/local: running validation operation
2020/01/02 14:42:32 [INFO] terraform: building graph: GraphTypeValidate
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.ConfigTransformer
2020/01/02 14:42:32 [TRACE] ConfigTransformer: Starting for path: 
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.ConfigTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.LocalTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.LocalTransformer with new graph:
  local.ci - *terraform.NodeLocal
  ------
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.OutputTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.OutputTransformer with new graph:
  local.ci - *terraform.NodeLocal
  output.ci - *terraform.NodeApplyableOutput
  ------
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.OrphanResourceInstanceTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.OrphanResourceInstanceTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.StateTransformer
2020/01/02 14:42:32 [TRACE] StateTransformer: state is empty, so nothing to do
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.StateTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.OrphanOutputTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.OrphanOutputTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.AttachResourceConfigTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.AttachResourceConfigTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.AttachStateTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.AttachStateTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.RootVariableTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.RootVariableTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.MissingProvisionerTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.MissingProvisionerTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.ProvisionerTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.ProvisionerTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.ModuleVariableTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.ModuleVariableTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.graphTransformerMulti
2020/01/02 14:42:32 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ProviderConfigTransformer
2020/01/02 14:42:32 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ProviderConfigTransformer with new graph:
  local.ci - *terraform.NodeLocal
  output.ci - *terraform.NodeApplyableOutput
  ------
2020/01/02 14:42:32 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.MissingProviderTransformer
2020/01/02 14:42:32 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.MissingProviderTransformer (no changes)
2020/01/02 14:42:32 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ProviderTransformer
2020/01/02 14:42:32 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ProviderTransformer (no changes)
2020/01/02 14:42:32 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.PruneProviderTransformer
2020/01/02 14:42:32 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.PruneProviderTransformer (no changes)
2020/01/02 14:42:32 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ParentProviderTransformer
2020/01/02 14:42:32 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ParentProviderTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.graphTransformerMulti (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.RemovedModuleTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.RemovedModuleTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.AttachSchemaTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.AttachSchemaTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.ReferenceTransformer
2020/01/02 14:42:32 [DEBUG] ReferenceTransformer: "local.ci" references: []
2020/01/02 14:42:32 [DEBUG] ReferenceTransformer: "output.ci" references: [local.ci]
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.ReferenceTransformer with new graph:
  local.ci - *terraform.NodeLocal
  output.ci - *terraform.NodeApplyableOutput
    local.ci - *terraform.NodeLocal
  ------
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.CountBoundaryTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.CountBoundaryTransformer with new graph:
  local.ci - *terraform.NodeLocal
  meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    local.ci - *terraform.NodeLocal
    output.ci - *terraform.NodeApplyableOutput
  output.ci - *terraform.NodeApplyableOutput
    local.ci - *terraform.NodeLocal
  ------
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.TargetsTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.TargetsTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.ForcedCBDTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.ForcedCBDTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.CloseProviderTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.CloseProviderTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.CloseProvisionerTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.CloseProvisionerTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.RootTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.RootTransformer (no changes)
2020/01/02 14:42:32 [TRACE] Executing graph transform *terraform.TransitiveReductionTransformer
2020/01/02 14:42:32 [TRACE] Completed graph transform *terraform.TransitiveReductionTransformer with new graph:
  local.ci - *terraform.NodeLocal
  meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    output.ci - *terraform.NodeApplyableOutput
  output.ci - *terraform.NodeApplyableOutput
    local.ci - *terraform.NodeLocal
  ------
2020/01/02 14:42:32 [DEBUG] Starting graph walk: walkValidate
2020/01/02 14:42:32 [TRACE] dag/walk: updating graph
2020/01/02 14:42:32 [TRACE] dag/walk: added new vertex: "local.ci"
2020/01/02 14:42:32 [TRACE] dag/walk: added new vertex: "output.ci"
2020/01/02 14:42:32 [TRACE] dag/walk: added new vertex: "meta.count-boundary (EachMode fixup)"
2020/01/02 14:42:32 [TRACE] dag/walk: added edge: "output.ci" waiting on "local.ci"
2020/01/02 14:42:32 [TRACE] dag/walk: added edge: "meta.count-boundary (EachMode fixup)" waiting on "output.ci"
2020/01/02 14:42:32 [TRACE] dag/walk: dependencies changed for "output.ci", sending new deps
2020/01/02 14:42:32 [TRACE] dag/walk: dependencies changed for "meta.count-boundary (EachMode fixup)", sending new deps
2020/01/02 14:42:32 [TRACE] dag/walk: visiting "local.ci"
2020/01/02 14:42:32 [TRACE] vertex "local.ci": starting visit (*terraform.NodeLocal)
2020/01/02 14:42:32 [TRACE] vertex "local.ci": evaluating
2020/01/02 14:42:32 [TRACE] [walkValidate] Entering eval tree: local.ci
2020/01/02 14:42:32 [TRACE] <root>: eval: *terraform.EvalLocal
2020/01/02 14:42:32 [ERROR] <root>: eval: *terraform.EvalLocal, err: Error in function call: Call to function "yamldecode" failed: cannot parse "0ba" as tag:yaml.org,2002:int.
2020/01/02 14:42:32 [TRACE] [walkValidate] Exiting eval tree: local.ci
2020/01/02 14:42:32 [TRACE] vertex "local.ci": visit complete
2020/01/02 14:42:32 [TRACE] dag/walk: upstream of "output.ci" errored, so skipping
2020/01/02 14:42:32 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020/01/02 14:42:32 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2020/01/02 14:42:32 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock

Error: Error in function call

  on test.tf line 2, in locals:
   2:   ci = yamldecode("key: 0ba")

Call to function "yamldecode" failed: cannot parse "0ba" as
tag:yaml.org,2002:int.
```

### Expected Behavior
The 0ba should be parsed as string as our expected

### Actual Behavior
$ terraform1218 apply                                                                                                                                                                                                                

Error: Error in function call

  on test.tf line 2, in locals:
   2:   ci = yamldecode("key: 0ba")

Call to function "yamldecode" failed: cannot parse "0ba" as
tag:yaml.org,2002:int.


### Steps to Reproduce
1. use that terraform file to apply

### Additional Context
<!--
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?
-->

### References

--- a/go.mod
+++ b/go.mod
@@ -121,7 +121,7 @@ require (
 	github.com/xiang90/probing v0.0.0-20160813154853-07dd2e8dfe18 // indirect
 	github.com/xlab/treeprint v0.0.0-20161029104018-1d6e34225557
 	github.com/zclconf/go-cty v1.5.0
-	github.com/zclconf/go-cty-yaml v1.0.1
+	github.com/zclconf/go-cty-yaml v1.0.2
 	go.uber.org/atomic v1.3.2 // indirect
 	go.uber.org/multierr v1.1.0 // indirect
 	go.uber.org/zap v1.9.1 // indirect

--- a/lang/functions_test.go
+++ b/lang/functions_test.go
@@ -978,6 +978,12 @@ func TestFunctions(t *testing.T) {
 				`yamldecode("true")`,
 				cty.True,
 			},
+			{
+				`yamldecode("key: 0ba")`,
+				cty.ObjectVal(map[string]cty.Value{
+					"key": cty.StringVal("0ba"),
+				}),
+			},
 		},
 
 		"yamlencode": {

